sgkit/stats/popgen.py

```diff
@@ -583,15 +586,21 @@ def Tajimas_D(
     c2 = b2 - ((n + 2) / (a1 * n)) + (a2 / (a1 ** 2))
     e1 = c1 / a1
     e2 = c2 / (a1 ** 2 + a2)
-    d_stdev = np.sqrt((e1 * S) + (e2 * S * (S - 1)))
-
-    if d_stdev == 0:
-        D = np.nan
-    else:
-        # finally calculate Tajima's D
+    d_stdev = da.sqrt((e1 * S) + (e2 * S * (S - 1)))
+
+    # FIXME there doesn't appear to be a way to do the np.errstate trick
+    # below in Dask (https://github.com/dask/dask/issues/3245) so we call
+    # compute() and turn the values into numpy arrays. Since this is a simple
+    # computation this is unlikely to be an issue in practical terms, but
+    # may lead to surprises for users if they expect all the computations
+    # done in sgkit to be lazy.
+    d = np.array(d.compute())
+    d_stdev = np.array(d_stdev.compute())
+    with np.errstate(invalid="ignore", divide="ignore"):
```

Comment:
The Dask issue is about controlling warnings, rather than the computation not working, so I wonder if we could remove the `with` block completely, and live with the warnings. I think that would be preferable to the eager computation.

To make the test pass, you can just add the following to `test_Tajimas_D`:

```python
@pytest.mark.filterwarnings("ignore::RuntimeWarning")
```